File: typescript/src/adapters/a2a/agents/events.ts
/**
 * Copyright 2025 © BeeAI a Series of LF Projects, LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { Callback } from "@/emitter/types.js";

export interface A2AAgentUpdateEvent {
  value: any;
}

export interface A2AAgentErrorEvent {
  message: string;
}

export interface A2AAgentEvents {
  update?: Callback<A2AAgentUpdateEvent>;
  error?: Callback<A2AAgentErrorEvent>;
}

File: python/beeai_framework/adapters/agentstack/agents/events.py
# Copyright 2025 © BeeAI a Series of LF Projects, LLC
# SPDX-License-Identifier: Apache-2.0

from pydantic import BaseModel, ConfigDict

try:
    import a2a.client as a2a_client
    import a2a.types as a2a_types
except ModuleNotFoundError as e:
    raise ModuleNotFoundError(
        "Optional module [agentstack] not found.\nRun 'pip install \"beeai-framework[agentstack]\"' to install."
    ) from e


class AgentStackAgentUpdateEvent(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)

    value: a2a_client.ClientEvent | a2a_types.Message


class AgentStackAgentErrorEvent(BaseModel):
    message: str


agent_stack_agent_event_types: dict[str, type] = {
    "update": AgentStackAgentUpdateEvent,
    "error": AgentStackAgentErrorEvent,
}
